### DISCUSSION TITLE
ObjectInstance.ToString() raises an JavaScriptException

### DISCUSSION THREAD

**Discussion by dmsch** [sebastienros/jint#981] [Q&A] [answered]
I'm trying to create Javascript objects in C# and use them as arguments when using Invoke(myObject), and I noticed that objectInstance.ToString() throws an exception:

```C#
var engine = new Engine();
var instance = new ObjectInstance(engine);
Console.WriteLine(instance.ToString()); // raises JavaScriptException
```

Is this a bug or do I need to use a different class to create Javascript objects in C#?

**Comment by lahma** [comment, marked as answer]
There problem here is that you create a plain `ObjectInstance` without a prototype chain. Prototype contains needed guidance how to handle common tasks, it also gives the JS functionality you are expected to find from any JS object. So two alternatives:

Use the way how engine constructs the final object (preferred):

```csharp
using System;
using Jint;
using Jint.Runtime;

var engine = new Engine();

var instance = engine.Realm.Intrinsics.Object.Construct(Arguments.Empty);
Console.WriteLine(instance.ToString());
```

Or set prototype manually:

```csharp
using System;
using Jint;
using Jint.Native.Object;

var engine = new Engine();

var instance = new ObjectInstance(engine);
instance.SetPrototypeOf(engine.Realm.Intrinsics.Object.PrototypeObject);
Console.WriteLine(instance.ToString());
```

**Reply by lahma** [reply]
Now that I think it through, it's a bit it isn't the default. Seems like a safe default at least.

**Reply by dmsch** [reply]
Thanks for your detailed answer.

**Reply by lahma** [reply]
You're welcome. I'll check in the near future if we can fix the default so your sample would work out of the box.

**Reply by lahma** [reply]
This is now fixed in main and available in MyGet feed.

**Reply by dmsch** [reply]
Excellent, thanks for the efforts.
